cocotb/triggers.py

```diff
@@ -194,7 +194,19 @@ def __init__(self, time_ps, units=None):
 
         See Also:
             :func:`~cocotb.utils.get_sim_steps`
+
+        Raises:
+            TriggerException: If a negative value is passed for Timer setup.
+
+        .. versionadded:: 1.5
+            Raise an exception when Timer uses a negative value as it is undefined behavior.
+            Warn for 0 as this will cause erratic behavior in some simulators as well.
         """
+        if time_ps <= 0:
+            if time_ps == 0:
+                self.log.warning("Timer setup with value 0, which might exhibit undefined behavior in some simulators")
+            else:
+                raise TriggerException("Timer value time_ps must not be negative")
```

Comment:
On that note, it would be good to add a test for this too

Reply:
Done.
![grafik](https://user-images.githubusercontent.com/25175069/95468534-e8ba8800-097e-11eb-98fa-33109130b862.png)